File: iSOLID.RiskAdvisory.Domain/Model/Tab.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace iSOLID.RiskAdvisory.Domain.Model
{
    public class Tab : Entity
    {
        public Tab()
        {
            this.Controls = new List<Control>();
        }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; }

        public List<Control> Controls { get; set; }
    }
}

File: iSOLID.RiskAdvisory.App/wwwroot/js/models/tab.js
leala.models.tab = function (d) {

    var self = this;

    self.name = ko.observable().extend({ required: true, maxLength: 100 });
    self.controls = ko.observableArray();

    if (d) {
        self.name(d.name);
        self.controls.removeAll();
        var controls = new Array();
        for (var i = 0; i < d.controls.length; i++) {
            var control = new leala.models.tab(d.controls[i]);
            controls.push(control);
        }
        self.controls(controls);
    }
}